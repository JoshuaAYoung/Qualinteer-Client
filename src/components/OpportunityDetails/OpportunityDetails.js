import React, { Component } from 'react';
import VableContext from '../../context/VableContext'
import { cleanDate, LinkName } from '../../utilities/Utils'
import VolunteerButton from '../VolunteerButton/VolunteerButton'

export default class OpportunityDetails extends Component {
  static contextType = VableContext;

  render() {
    const { title, description, contact, start_date, duration, commitment, ed_level, experience, license, remote, posted, name, address, city, state, zipcode, phone, website } = this.context.opportunity
    return (
      <>
        <h3 className='opportunityTitle'>{title}</h3>
        <LinkName name={name} website={website} />
        <p className='orgInfo'>{address}</p>
        <p className='orgInfo'>{city}, {state} {zipcode}</p>
        <p className='orgInfo'>{phone}</p>
        <VolunteerButton
          title={title}
          contact={contact}
          name={name}
          history={this.props.history}
        />
        <p className='opportunityInfo'><b>POSTED ON: </b>{cleanDate(posted)}</p>
        <p className='opportunityInfo'><b>START DATE: </b>{start_date}</p>
        <p className='opportunityInfo'><b>JOB DURATION: </b>{duration}</p>
        <p className='opportunityInfo'><b>JOB COMMITMENT (HRS/WK): </b>{commitment}</p>
        <p className='opportunityInfo'><b>EDUCATION LEVEL REQUESTED: </b>{ed_level}</p>
        {experience && <p className='opportunityInfo'><b>EXPERIENCE REQUESTED: </b>{experience}</p>}
        {license && <p className='opportunityInfo'><b>LICENSE REQUESTED: </b>{license}</p>}
        <p className='opportunityInfo'><b>OPTION TO VOLUNTEER REMOTELY: </b>{remote ? 'Yes' : 'No'} </p>
        <p className='opportunityDescription'><b>JOB DESCRIPTION: </b></p><div dangerouslySetInnerHTML={{ __html: description }} />
      </>
    );
  }
}