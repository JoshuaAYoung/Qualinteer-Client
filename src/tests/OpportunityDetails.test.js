import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import OpportunityDetails from '../components/OpportunityDetails/OpportunityDetails';
import VableContext from '../context/VableContext';

describe('OpportunityDetails Component', () => {
  const opportunity = {
    title: 'test title',
    description: 'test description',
    contact: 'test contact',
    start_date: 'test start_date',
    duration: 'test duration',
    commitment: 'test commitment',
    ed_level: 'none',
    experience: 'test experience',
    license: 'test license',
    remote: false,
    posted: '2019-12-22T16:28:32.615Z',
    name: 'test name',
    address: 'test address',
    city: 'test city',
    state: 'test state',
    zipcode: 90210,
    phone: 'test phone',
    website: 'test website',
  };

  it('OpportunityDetails Smoke Test', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <VableContext.Provider value={{ opportunity }}>
        <OpportunityDetails />
      </VableContext.Provider>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('OpportunityDetails Snapshot Test', () => {
    const wrapper = shallow(
      <VableContext.Provider value={{ opportunity }}>
        <OpportunityDetails />
      </VableContext.Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
