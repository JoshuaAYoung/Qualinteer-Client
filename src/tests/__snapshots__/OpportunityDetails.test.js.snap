// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpportunityDetails Component OpportunityDetails Snapshot Test 1`] = `<OpportunityDetails />`;
